platform/api/akamai/instance.go

```diff
@@ -0,0 +1,186 @@
+// Copyright The Mantle Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package akamai
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/linode/linodego"
+
+	"github.com/coreos/pkg/capnslog"
+)
+
+var (
+	plog = capnslog.NewPackageLogger("github.com/flatcar/mantle", "platform/api/akamai")
+	tags = []string{"mantle"}
+)
+
+// Server is a wrapper around Akamai instance Server struct.
+type Server struct {
+	*linodego.Instance
+}
+
+// CreateServer is running a few steps:
+// 1. Create the linode (i.e) the instance but it does not boot it
+// 2. Create a custom disk that uses the Flatcar image previously uploaded
+// 3. Resize the disk to match the expected disk size of the instance
+// 4. Create a custom configuration to directly boot on the disk
+// 5. Boot the instance
+func (a *API) CreateServer(ctx context.Context, name, userData string) (*Server, error) {
+	booted := false
+	opts := linodego.InstanceCreateOptions{
+		Label:  name,
+		Region: a.opts.Region,
+		Type:   a.opts.Type,
+		Tags:   tags,
+		Metadata: &linodego.InstanceMetadataOptions{
+			UserData: base64.StdEncoding.EncodeToString([]byte(userData)),
+		},
+		Booted:    &booted,
+		PrivateIP: true,
+	}
+
+	plog.Infof("Creating the instance")
+	instance, err := a.client.CreateInstance(ctx, opts)
+	if err != nil {
+		return nil, fmt.Errorf("creating instance: %w", err)
+	}
+
+	t, err := a.client.GetType(ctx, a.opts.Type)
+	if err != nil {
+		return nil, fmt.Errorf("getting instance type: %w", err)
+	}
+
+	// RootPass is required when you create a disk from an image. It is not used by Flatcar.
+	instanceDiskCreateOptions := linodego.InstanceDiskCreateOptions{
+		Size:     t.Disk,
+		Label:    name,
+		Image:    a.opts.Image,
+		RootPass: "ThisPasswordIsNotUsedButRequiredByAkamai1234",
```

Comment:
You wrote that it is not used by Flatcar, so can I assume that this root password won't be used as a password for the root user in the booted instance?